src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -61,6 +67,44 @@ public void Configure(string name, ScriptApplicationHostOptions options)
                 options.IsSelfHost = options.IsSelfHost;
                 options.IsStandbyConfiguration = true;
             }
+
+            options.IsFileSystemReadOnly = IsZipDeployment(options);
+        }
+
+        private bool BlobExists()
+        {
+            return _cloudBlockBlobHelperService.BlobExists(_environment.GetEnvironmentVariable(ScmRunFromPackage)).GetAwaiter().GetResult();
```

Comment:
We shouldn't be doing sync over async like this. I don't understand why we need to be checking for actual blob existence like this, particularly on startup which will affect coldstart.

Why isn't the existing EnvironmentSettings.IsZipDeployment sufficient?

Reply:
Copying from PR description:

**It's not sufficient to check if SCM_RUN_FROM_PACKAGE is set, the URI also needs to point to an existing blob to deduce it's a zip deployment (therefore read-only).** 

This is an unfortunate limitation as of today - as this app setting is needed for Kudu deployments.

Reply:
>  particularly on startup which will affect coldstart.

Some things to note:
1) In prod today, we already have a check for blob existence during specialization. It happens in [this call](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Management/InstanceManager.cs#L305).
2) The newly added check will only happen once and be cached, so the check in 1) won't be calling storage again.
3) The call for actual blob existence only happens if `SCM_RUN_FROM_PACKAGE` is present, which is true for Linux Consumption.

Reply:
That check only happens during Linux consumption, but the change you're making here will be for all platforms. Problems with the above call aside, I still don't see why what we have now isn't sufficient. The existing IsZipDeployment logic will only return true if the env var is present AND it's a URI format or 1, which is correct. Can you update the bug with details on when the existing check doesn't work? I.e. what value do the settings have.

Reply:
@mathewc BlobExists() will only be called if
IsValidZipSetting(_environment.GetEnvironmentVariable(ScmRunFromPackage) is true

since this will be true only on Linux consumption we are still be retaining the old behavior

On Linux consumption ScmRunFromPackage will always point to an external url. But preference is given to WEBSITE_RUN_FROM_PACAKGE if present.

So currently:
BlobExists() will be called only on LinuxConsumption and only if WEBSITE_RUN_FROM_PACAKGE is empty

With this change as well I think we have the same behavior.

Reply:
I see. I was unaware we had that logic already and that you're moving things around/fixing. Let's at least make the call sync here then, to avoid the sync over async ani-pattern which has caused us problems in the past (more info [here](https://devblogs.microsoft.com/pfxteam/should-i-expose-synchronous-wrappers-for-asynchronous-methods/).

Reply:
Changed `BlobExists` to sync, thanks for the suggestions.